test/UnitTest/Microsoft.AspNetCore.OData.Test/Extensions/Endpoint/ODataEndpointRouteValueTransformerTests.cs

```diff
@@ -104,6 +107,96 @@ public void TransformAsyncReturnsCorrectRouteValues()
             Assert.Same(actionDescriptor, httpContext.ODataFeature().ActionDescriptor);
         }
 
+        [Theory]
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        public void TransformAsyncReturnsCorrectRouteValuesForNonOptionRequests(bool isOptionsRequest, bool includeMetadata)
+        {
+            // Arrange
+            IEndpointRouteBuilder builder = EndpointRouteBuilderFactory.Create("odata",
+                c => c.AddService(ServiceLifetime.Singleton, b => GetEdmModel()));
+
+            HttpContext httpContext = new DefaultHttpContext
+            {
+                RequestServices = builder.ServiceProvider
+            };
+
+            var httpMethod = isOptionsRequest ? HttpMethod.Options : HttpMethod.Get;
+
+            HttpRequest request = SetHttpRequest(httpContext, httpMethod, "http://localhost:123/Customers(1)");
+            RouteValueDictionary values = new RouteValueDictionary();
+            values.Add("ODataEndpointPath_odata", "Customers(1)");
+
+            ActionDescriptor actionDescriptor = new ControllerActionDescriptor
+            {
+                ControllerName = "Customers",
+                ActionName = "Get",
+                EndpointMetadata = GenerateMetadata(includeMetadata)
+            };
+
+            IActionSelector actionSelector = new MockActionSelector(actionDescriptor);
+
+            // Act
+            ODataEndpointRouteValueTransformer transformer = new ODataEndpointRouteValueTransformer(actionSelector);
+            ValueTask<RouteValueDictionary> actual = transformer.TransformAsync(httpContext, values);
```

Comment:
```suggestion
            _ = transformer.TransformAsync(httpContext, values);
```
Since `actual` is assigned but never used maybe this might be better?

Also do you need to Wait for `TransformAsync`?

Reply:
To be sure I made the methods use async await